subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -28,6 +28,32 @@ static const uint16_t valid_5g_chans_20mhz[] = {32, 36, 40, 44, 48, 52, 56, 60,
 	104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 159, 161,
 	163, 165, 167, 169, 171, 173, 175, 177};
 
+
+static char *wifi_utils_strdup(const char *orig_str)
+{
+	char *dup_str;
+	int len;
+
+	if (!orig_str) {
+		NET_ERR("Cannot process NULL string");
+		return NULL;
+	}
+
+	len = strlen(orig_str);
+
+	dup_str = k_malloc(len + 1);
```

Comment:
This introduces the dependency of `CONFIG_HEAP_MEM_POOL_SIZE` being non-zero and enough to cover the string. And as we minimize the dynamic memory usage inside Zephyr kernel, may be we should switch to using a static buffer with max size? I guess 8 bytes for bands and 32 * MAX_SSIDS for SSIDs?

Reply:
> may be we should switch to using a static buffer with max size? I guess 8 bytes for bands and 32 * MAX_SSIDS for SSIDs

I would actually prefer not to use malloc and friends here, we have tried to use static buffers in networking code and it has worked quite well earlier.

Reply:
I think that is fine, the buffer sizes can be adjusted if needed.